### DISCUSSION TITLE
Recording without Client connected possible?

### DISCUSSION THREAD

**Discussion by bithand** [bluenviron/mediamtx#2866] [Questions] [answered]
### Question

First of all, I would like to thank the authors for this great tool. It's like a Swiss army knife for a variety of streaming applications!

In the following I have a question of understanding:
In the following example, I only have an RTSP camera that serves as the source and whose stream is to be recorded. This happens with : 

`source: rtsp://admin:xxxxxx@192.168.55.78/H.264`

To record the data, the recording properties are set as follows:

```
record: yes
recordPath: ./recordings/%path/%Y-%m-%d_%H-%M-%S
recordFormat: fmp4
recordPartDuration: 1s
recordSegmentDuration: 1m
recordDeleteAfter: 1h
```

Since my source is a URL, I have also set the following property:

`sourceOnDemand: yes`

Now to my question. I would like to start a recording even without a client being connected to view it. As I have understood it so far, I would have to use a "trigger" for this, and capture and save the stream there with ffmeg, for example. The question would simply be whether there is an easier way to do this, or whether there is perhaps already an option for this?

Thanks in advance and best regards!

**Comment by aler9** [comment, marked as answer]
Hello, 

> it's like a Swiss army knife for a variety of streaming applications!

thanks, this was exactly the purpose :D

> Since my source is a URL, I have also set the following property: sourceOnDemand: yes

sourceOnDemand is not mandatory in any way unless you want to save bandwidth and pull a remote stream exclusively when there are readers connected to it, which is not the case, since you want 24/7 recording. Set it to `no` and you'll get exactly what you want.

**Reply by bithand** [reply]
Thank you very much for your quick reply. 
After your answer, I simply read the error message thoroughly :-) It turned out that the sourceOnDemand option requires a dedicated path and on the default path it must always be 'on' when a URL is specified as source. This was simply not clear to me. 
Thanks for your advice!

[a comment by an automated account is not shown]
